### ISSUE TITLE
Blank/Null values now allowed in templated New-Azresourcegroupdeployment parameters

### ISSUE THREAD

**Issue by marshalexander99** [Azure/azure-powershell#19923]
### Description

With the following code

`New-AzResourceGroupDeployment -ResourceGroupName $rgname -TemplateFile $templatefile  -storage_account_name $storage_account_name  -storage_account_kind $storage_account_kind  -storage_account_sku $storage_account_sku  -container_name $container_name  -avdvnetname $avdvnetname  -avdvnetrg $avdvnetrg  -avdvnetsub $avdvnetsub  -allowedip1 $allowedip`

if $allowedip is set to null or is left blank command doesn't run.

Parameter should be allowed to be blank or empty if an allowed IP is not required as bicep template allows for the following

`resource storage_account 'Microsoft.Storage/storageAccounts@2022-05-01' = {
  location: location
  tags: resourceGroup().tags
  kind: storage_account_kind
  name: storage_account_name
  sku: {
    name: storage_account_sku
  }
  properties: {
    allowBlobPublicAccess: false
    minimumTlsVersion: 'TLS1_2'
    publicNetworkAccess: 'Disabled'
    networkAcls: {
      defaultAction: 'Deny'
      bypass: 'AzureServices'
      virtualNetworkRules:[
        {
          action:'Allow'
          id:avdvnet_resource.id
        }
      ]
      ipRules: empty(allowedip1) ? [
        {
          action:'Allow'
          value:allowedip1
        }
      ] : null
    }
  }
}`

### Issue script & Debug output

```PowerShell
New-AzResourceGroupDeployment : Cannot bind argument to parameter 'allowedip1' because it is an empty string.
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      5.1.17763.771
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.771
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Module versions

```PowerShell
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     2.10.1     Az.Accounts                         {Add-AzEnvironment, Clear-AzConfig, Clear-AzContext, Clear...
Script     6.2.0      Az.Resources                        {Export-AzResourceGroup, Export-AzTemplateSpec, Get-AzDeny...
Script     4.9.0      Az.Storage
```

### Error output

New-AzResourceGroupDeployment : Cannot bind argument to parameter 'allowedip1' because it is an empty string.

**Comment by dingmeng-xue** [comment]
Thanks for reporting. We will try to reproduce this issue. The question from our side is whether you can remove `-allowedip1` if it is empty?

**Comment by marshalexander99** [comment]
The point of having a deployment script is to allow for values to be added in as required. There shouldn't need to be a modification to remove the parameter should it not be required.
Currently I have to set the IP address to 0.0.0.0 if it's to be excluded and utilise some additional logic within the bicep code to ignore it. It would just be easier if it would allow null/blank values

**Comment by Nickcandy** [comment]
@marshalexander99  Hi, for now I've tested it's not available if we set allowedip to $null, but we can set it to '', like this:
`$allowedip=''`
`New-AzResourceGroupDeployment -ResourceGroupName $rgname -TemplateFile $templatefile -storage_account_name $storage_account_name -storage_account_kind $storage_account_kind -storage_account_sku $storage_account_sku -allowedip1 $allowedip -location $location`
I wonder if it's a solution for you. Cause we cannot customize for only one parameter for now.
Besides, in your bicep, there is a mistake, which should be:
`			ipRules: empty(allowedip1) ? null : [ 
				{ 
					action:'Allow' 
					value:allowedip1 
				} 
			] } `
This mistake may also lead to this error.
